docs/www/nodule-grid.js

```diff
@@ -377,6 +377,368 @@
                 $("#add-first-nodule").toggle(count === 0);
             }
 
+            function _revertData() {
+                // put back original data where it was saved from an updateData call
+                //VAPALS.displayNotification("Reverting Data");
+                if (!confirm("Do you really want to revert to the data prior to the data update?")) {
+                    return;
+                }
+
+                // should breadcrumbs be saved in added nodules and then these be removed - not for now
+                
+                // loop thru all potential nodules even if not shown since they may get hidden after a data update
+                //for (let noduleId = 1; noduleId <= settings.getNoduleCount(); noduleId++) {
+                for (let noduleId = 1; noduleId <= options.availableNodules; noduleId++) {
```

Comment:
This entire function needs to be greatly simplified. There is no need to iterate over each nodule, nor to look at individual fields. The fields you want to revert will always have the "original-value" class. You can get a list of all these fields using the most common jquery selector: `$("#nodule-table").find("[original-value]")`. With that list, write some code that removes the classes, and sets the value back to the original. The only thing you'll have to consider is if that input control is a text input or a dropdown (setting values on these two types of inputs is slightly different).